currencies/rate_converter_test.go

```diff
@@ -13,354 +13,273 @@ import (
-}
-
-func TestFetch_FailErrorHttpClient(t *testing.T) {
-
-	// Setup:
-	calledURLs := []string{}
-	mockedHttpServer := httptest.NewServer(http.HandlerFunc(
-		func(rw http.ResponseWriter, req *http.Request) {
-			calledURLs = append(calledURLs, req.RequestURI)
-			rw.WriteHeader(http.StatusBadRequest)
-		}),
-	)
-
-	defer mockedHttpServer.Close()
-
-	// Execute:
-	currencyConverter := currencies.NewRateConverter(
-		&http.Client{},
-		mockedHttpServer.URL,
-		time.Duration(24)*time.Hour,
-	)
-
-	// Verify:
-	assert.Equal(t, 1, len(calledURLs), "sync URL should have been called %d times but was %d", 1, len(calledURLs))
-	assert.Equal(t, currencyConverter.LastUpdated(), (time.Time{}), "LastUpdated() shouldn't return a time set")
-	assert.Nil(t, currencyConverter.Rates(), "Rates() should return nil")
+			"GBP":{
+				"USD":1.2952
+			}
+		}
+	}`)
 }
 
-func TestFetch_FailBadSyncURL(t *testing.T) {
-
-	// Setup:
-
-	// Execute:
-	currencyConverter := currencies.NewRateConverter(
-		&http.Client{},
-		"justaweirdurl",
-		time.Duration(24)*time.Hour,
-	)
-
-	// Verify:
-	assert.Equal(t, currencyConverter.LastUpdated(), (time.Time{}), "LastUpdated() shouldn't return a time set")
-	assert.Nil(t, currencyConverter.Rates(), "Rates() should return nil")
+func formatTestErrorMsg(desc string, message string, args ...interface{}) string {
+	message = fmt.Sprintf(message, args)
+	return fmt.Sprintf("Test Case: %s \n Message: %s", desc, message)
 }
 
-func TestFetch_FailBadJSON(t *testing.T) {
-
+func TestNewRateConverter(t *testing.T) {
 	// Setup:
-	calledURLs := []string{}
-	mockedHttpServer := httptest.NewServer(http.HandlerFunc(
-		func(rw http.ResponseWriter, req *http.Request) {
-			calledURLs = append(calledURLs, req.RequestURI)
-			rw.WriteHeader(http.StatusOK)
-			rw.Write([]byte(
-				`{
-					"dataAsOf":"2018-09-12",
-					"conversions":{
-						"USD":{
-							"GBP":0.77208
-						},
-						"GBP":{
-							"USD":1.2952
-						},
-						"badJsonHere"
-					}
-				}`,
-			))
-		}),
-	)
-
-	defer mockedHttpServer.Close()
+	mockServerHandler := func(mockResponse []byte, code int) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			rw.WriteHeader(code)
+			rw.Write(mockResponse)
+		})
+	}
 
-	// Execute:
-	currencyConverter := currencies.NewRateConverter(
-		&http.Client{},
-		mockedHttpServer.URL,
-		time.Duration(24)*time.Hour,
-	)
+	testCases := []struct {
+		mockResponse     []byte
+		mockResponseCode int
+		updateInterval   int // In seconds
+		fetchURL         string
+		expectedTicks    int
+		description      string
+	}{
+		{
+			mockResponse:     getMockRates(),
+			mockResponseCode: 200,
+			updateInterval:   1,
+			expectedTicks:    1,
+			description:      "Fetching currency rates successfully",
+		},
+		{
+			mockResponse:     []byte(""),
+			mockResponseCode: 200,
+			updateInterval:   100,
+			description:      "Currency rates endpoint returns empty response",
```

Comment:
What's the expectation for the test? If it's 0 ticks, consider still including that for readability.. even though it's not necessary per the compiler.